src/main/java/appeng/api/networking/security/BaseActionSource.java

```diff
@@ -13,13 +13,29 @@
 
 package appeng.api.networking.security;
 
-public class BaseActionSource {
+/**
+ * Represents the source of an action within the network.
+ */
+public interface BaseActionSource {
```

Comment:
Changes like this in `appeng/api` shouldn't be made lightly, this will affect a lot of mods both inside and outside of NH, which we must still support since AE2 is active support outside of the pack